src/shims/unix/macos/foreign_items.rs

```diff
@@ -174,18 +174,16 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 // are met, then the name is set and 0 is returned. Otherwise, if
                 // the specified name is lomnger than MAXTHREADNAMESIZE, then
                 // ENAMETOOLONG is returned.
-                //
-                // FIXME: the real implementation maybe returns ESRCH if the thread ID is invalid.
                 let thread = this.pthread_self()?;
-                let res = if this.pthread_setname_np(
+                let res = match this.pthread_setname_np(
                     thread,
                     this.read_scalar(name)?,
                     this.eval_libc("MAXTHREADNAMESIZE").to_target_usize(this)?.try_into().unwrap(),
                     /* truncate */ false,
                 )? {
-                    Scalar::from_u32(0)
-                } else {
-                    this.eval_libc("ENAMETOOLONG")
+                    ThreadNameResult::Ok => Scalar::from_u32(0),
+                    ThreadNameResult::NameTooLong => this.eval_libc("ENAMETOOLONG"),
+                    ThreadNameResult::ThreadNotFound => this.eval_libc("ESRCH"),
```

Comment:
Where did you get the ESRCH from? I don't see it in the code links above.

Reply:
From the source code (:

See https://github.com/rust-lang/miri/issues/3965#issuecomment-2449413110.

Reply:
Miri's source code needs to be self-explaining. So please include sufficient references to explain the errors codes.

Reply:
Should I write it as a comment? The link to the original source is here already.

Reply:
I didn't see the ESRCH in the original source though, did I look in the wrong place?

Reply:
Ah, sorry, that's about setting the name. Should be unreachable then, there's no way to set the name for a thread other than the current.

Reply:
Yes for getname I can see the ESRCH :)